### PR TITLE
Handled citizen action creation

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

File: app/config/services/platform.xml [hunk 1]
```diff
@@ -226,6 +226,10 @@
         </service>
 
         <!-- Citizen projects -->
+        <service id="app.citizen_project.manager" class="AppBundle\CitizenProject\CitizenProjectManager">
```
Comment (thomasbisignani) [inline app/config/services/platform.xml]: This service is needed ? you can use action injection (enabled in `services.xml`)

Comment (thomasbisignani) [reply app/config/services/platform.xml]: I think this service is not conditionally used. He's used in the `return` statement : `$citizenProjectManager->getCitizenProjectAdministrators($project)`

Comment (HeahDude) [reply app/config/services/platform.xml]: Yes you're right. Good catch, thanks!

File: src/Controller/EnMarche/CitizenActionManagerController.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * @Entity("action", expr="repository.findOneBySlug(slug)")
+ */
+class CitizenActionManagerController extends Controller
+{
+    /**
+     * @Route("/creer", name="app_citizen_action_manager_create", defaults={"slug": ""})
+     * @Method("GET|POST")
+     * @Security("is_granted('CREATE_CITIZEN_ACTION', project)")
+     */
+    public function createAction(Request $request, CitizenProject $project, CitizenProjectManager $citizenProjectManager): Response
```
Comment (Remg) [inline src/Controller/EnMarche/CitizenActionManagerController.php]: argument `CitizenProjectManager $citizenProjectManager` shouldn't be there

File: src/Controller/EnMarche/CitizenActionManagerController.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * @Route("/projets-citoyens/{project_slug}/actions")
+ * @Entity("project", expr="repository.findOneApprovedBySlug(project_slug)")
+ * @Entity("action", expr="repository.findOneBySlug(slug)")
+ */
+class CitizenActionManagerController extends Controller
+{
+    /**
+     * @Route("/creer", name="app_citizen_action_manager_create", defaults={"slug": ""})
```
Comment (Remg) [inline src/Controller/EnMarche/CitizenActionManagerController.php]: remove the default `slug` parameter

Comment (Remg) [inline src/Controller/EnMarche/CitizenActionManagerController.php]: remove default `slug` parameter

File: app/config/services/platform.xml [hunk 4]
```diff
@@ -368,6 +369,12 @@
             <argument type="service" id="app.repository.citizen_action"/>
         </service>
 
+        <service autowire="true" id="AppBundle\CitizenAction\CitizenActionCommandHandler">
+            <argument key="$mailer" type="service" id="app.mailer.transactional" />
```
Comment (B-Galati) [inline app/config/services/platform.xml]: Why not creating an alias for `app.mailer.transactional` and then use autowiring ?

File: src/Entity/CitizenAction.php [hunk 5]
```diff
@@ -52,9 +44,9 @@ public function __construct(
         $this->category = $category;
         $this->description = $description;
         $this->postAddress = $address;
-        $this->capacity = $capacity;
         $this->participantsCount = $participantsCount;
-        $this->beginAt = $beginAt;
+        // We need a \DateTime object for now to work with Gedmo sluggable
+        $this->beginAt = $beginAt instanceof \DateTimeImmutable ? new \DateTime($beginAt->format(\DATE_ATOM)) : $beginAt;
```
Comment (B-Galati) [inline src/Entity/CitizenAction.php]: Just Use https://secure.php.net/manual/en/datetimeimmutable.createfrommutable.php instead of `new \DateTime`

Comment (B-Galati) [reply src/Entity/CitizenAction.php]: For the record, I misundertood it's ok :-)

**Pull request merged**